hebi/CMakeLists.txt

```diff
@@ -14,6 +14,16 @@ if (UNIX AND NOT APPLE)
 
   set(LIBHEBI_TARGET_ARCHITECTURE "x86_64" CACHE STRING
     "Target architecture. [i686, x86_64, armhf, aarch64]")
+
+  # Detect cross-compilation, especially on the ROS build farm:
+  if ("i386-linux-gnu" STREQUAL ${CMAKE_LIBRARY_ARCHITECTURE})
+    set(LIBHEBI_TARGET_ARCHITECTURE "i686")
+  elseif ("arm-linux-gnueabihf" STREQUAL ${CMAKE_LIBRARY_ARCHITECTURE})
+    set(LIBHEBI_TARGET_ARCHITECTURE "armhf")
+  elseif ("aarch64-linux-gnu" STREQUAL ${CMAKE_LIBRARY_ARCHITECTURE})
+    set(LIBHEBI_TARGET_ARCHITECTURE "aarch64")
+  endif()
```

Comment:
For future edge case checking sake, want to add some more checks in case these change in the future? I could see `i686-linux-gnu` being an option and breaking stuff again

Reply:
Good point -- anything you can think of besides i686?

Reply:
If you look at this random package list from [Debian](https://packages.debian.org/jessie/i386/binutils-x86-64-linux-gnu/filelist), you can see a great mix of `i586` and `i386`. So I would just do a regex like `i[3456]86-linux-gnu`